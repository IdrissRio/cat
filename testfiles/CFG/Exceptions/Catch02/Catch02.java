public class Catch02 {
  void test() {
    try {
      try {
        m();
        int j;
      } catch (TestException e) {
        m();
      }
      int f;
    } catch (TestException e) {
      int c;
    }
    int d;
  }

  void m() throws TestException {}
}

class TestException extends RuntimeException {}
class TestSubException extends TestException {}
