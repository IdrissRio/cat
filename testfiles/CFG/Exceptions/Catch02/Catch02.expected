NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [MethodAccess[45]]
NodeName: MethodAccess[45] Successors: [VariableDeclarator[55], UncheckedExceptions[46]]
NodeName: VariableDeclarator[55] Successors: [UncheckedExceptions[73]]
NodeName: UncheckedExceptions[73] Successors: [VariableDeclarator[79], Exit[36]]
NodeName: VariableDeclarator[79] Successors: [UncheckedExceptions[99]]
NodeName: UncheckedExceptions[99] Successors: [VariableDeclarator[105], Exit[36]]
NodeName: VariableDeclarator[105] Successors: [Exit[36]]
NodeName: Exit[36] Successors: []
NodeName: UncheckedExceptions[46] Successors: [ParameterDeclaration[60], Exit[36]]
NodeName: ParameterDeclaration[60] Successors: [MethodAccess[67]]
NodeName: MethodAccess[67] Successors: [VariableDeclarator[79], UncheckedExceptions[68]]
NodeName: UncheckedExceptions[68] Successors: [ParameterDeclaration[84], Exit[36]]
NodeName: ParameterDeclaration[84] Successors: [VariableDeclarator[95]]
NodeName: VariableDeclarator[95] Successors: [VariableDeclarator[105]]
NodeName: Entry[109] Successors: [Exit[117]]
NodeName: Exit[117] Successors: []
NodeName: Entry[129] Successors: [Exit[130]]
NodeName: Exit[130] Successors: []
NodeName: Entry[121] Successors: [Exit[122]]
NodeName: Exit[122] Successors: []
NodeName: Entry[152] Successors: [Exit[153]]
NodeName: Exit[153] Successors: []
NodeName: Entry[144] Successors: [Exit[145]]
NodeName: Exit[145] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[36] Predecessor: [UncheckedExceptions[99], UncheckedExceptions[73], UncheckedExceptions[46], UncheckedExceptions[68], VariableDeclarator[105]]
NodeName: UncheckedExceptions[99] Predecessor: [VariableDeclarator[79]]
NodeName: VariableDeclarator[79] Predecessor: [UncheckedExceptions[73], MethodAccess[67]]
NodeName: UncheckedExceptions[73] Predecessor: [VariableDeclarator[55]]
NodeName: VariableDeclarator[55] Predecessor: [MethodAccess[45]]
NodeName: MethodAccess[45] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: MethodAccess[67] Predecessor: [ParameterDeclaration[60]]
NodeName: ParameterDeclaration[60] Predecessor: [UncheckedExceptions[46]]
NodeName: UncheckedExceptions[46] Predecessor: [MethodAccess[45]]
NodeName: UncheckedExceptions[68] Predecessor: [MethodAccess[67]]
NodeName: VariableDeclarator[105] Predecessor: [UncheckedExceptions[99], VariableDeclarator[95]]
NodeName: VariableDeclarator[95] Predecessor: [ParameterDeclaration[84]]
NodeName: ParameterDeclaration[84] Predecessor: [UncheckedExceptions[68]]
NodeName: Exit[117] Predecessor: [Entry[109]]
NodeName: Entry[109] Predecessor: []
NodeName: Exit[130] Predecessor: [Entry[129]]
NodeName: Entry[129] Predecessor: []
NodeName: Exit[122] Predecessor: [Entry[121]]
NodeName: Entry[121] Predecessor: []
NodeName: Exit[153] Predecessor: [Entry[152]]
NodeName: Entry[152] Predecessor: []
NodeName: Exit[145] Predecessor: [Entry[144]]
NodeName: Entry[144] Predecessor: []
